[jax2tf] Add converted primitives without tests to the generated doc.

bchetioui opened the pull request.

bchetioui committed:
[jax2tf] Add converted primitives without tests to the generated
doc.

gnecula commented on jax/experimental/jax2tf/primitives_with_limited_support.md:
```diff
@@ -75,3 +75,10 @@ conversion to Tensorflow.
+The following JAX primitives have a defined conversion but are known to be
+missing tests:
+
+`add_any`, `argmax`, `argmin`, `broadcast`, `call`, `clamp`, `complex`, `cond`, `conj`, `conv_general_dilated`, `custom_jvp_call_jaxpr`, `custom_lin`, `custom_vjp_call_jaxpr`, `dot_general`, `imag`, `integer_pow`, `real`, `remat_call`, `rev`, `scan`, `select_and_scatter`, `select_and_scatter_add`, `stop_gradient`, `transpose`, `xla_call`
```
xla_call, remat_call, call: these are not really seen by the converter, they are inlined. You should exclude them. Technically xla_pmap_p also. Search for xla_call_p in jax2tf.

scan, cond, while (why is it not on the list?): these are tested in control_flow_ops_test. 

I am surprised about select_and_scatter_add.

SIben replied:
`while` is called within the `random_gamma` conversion it seems.

bchetioui committed:
Ignore some primitives in the output of untested primitives.

gnecula commented on jax/experimental/jax2tf/tests/correctness_stats.py:
```diff
@@ -200,19 +205,35 @@ def _pipewrap(columns):
+  implemented but not in the set passed as an argument.
+  """
+  ignore = set([xla.xla_call_p, pxla.xla_pmap_p, pe.remat_call_p, core.call_p,
+                lax.cond_p, lax.scan_p, lax.while_p])
```
The "custom_xxx_call_jaxpr" are also well-tested, in the control-flow-ops-test. Perhaps remove them from here. Alternatively (may be even better) run both primitives_tests and control-flows tests when generating the documentation?

SIben replied:
Addressed in `7725d8db`; we now also run tests in `control_flow_ops_test` and this instruction is added to the template.

bchetioui committed:
Added control_flow_ops_test to template and updated primitives.

bchetioui committed:
Removed svd from the list of missing tests.

Was just included because I run the tests using
JAX_SKIP_SLOW_TESTS=1, which didn't run the SVD tests. Patched
the generated file manually.
